src/daemon/udp_connection/dlt_daemon_udp_socket.c

```diff
@@ -0,0 +1,262 @@
+        {
+            dlt_vlog(LOG_WARNING, "[%s/%d] Setsockopt error %s\n", __func__, __LINE__,
+                    strerror(errno));
+            continue;
+        }
+
+        if(setsockopt(*sock, SOL_SOCKET, SO_RCVBUF, &sockbuffer, DLT_DAEMON_RCVBUFSIZESOCK)
+                      == SYSTEM_CALL_ERROR)
+        {
+            dlt_vlog(LOG_WARNING, "[%s/%d] Setsockopt error %s\n", __func__, __LINE__,
+                    strerror(errno));
+            continue;
+        }
+
+        if (bind(*sock, addrinfo_iterator->ai_addr, addrinfo_iterator->ai_addrlen)
+                 == SYSTEM_CALL_ERROR) {
+            close(*sock);
+            dlt_log(LOG_WARNING, "bind() error\n");
+            continue;
+        }
+
+        break;
+    }
+
+    if (addrinfo_iterator == NULL) {
+        dlt_log(LOG_WARNING, "failed to bind socket\n");
+        return DLT_RETURN_ERROR;
+    }
+
+    freeaddrinfo(servinfo);
+
+    return DLT_RETURN_OK; // OK
+}
+
+// **************************************************************************
+// Function   : dlt_daemon_udp_dltmsg_multicast
+// In Param   : data bytes in dlt format
+// Out Param  : NIL
+// Description: multicast UDP dlt-message packets to dlt-client
+// **************************************************************************
+void dlt_daemon_udp_dltmsg_multicast( void* data1, int size1,
+                                                  void* data2, int size2,
+                                                  int verbose)
+{
+    PRINT_FUNCTION_VERBOSE(verbose);
+    dlt_daemon_udp_clientmsg_send(&g_udpmulticast_addr, data1, size1,
+                                  data2, size2, verbose);
+}
+
+// **************************************************************************
+// Function   : dlt_daemon_udp_clientmsg_send
+// In Param   : data bytes & respective size in dlt format
+// Out Param  : NIL
+// Description: common interface to send data via UDP protocol
+// **************************************************************************
+void dlt_daemon_udp_clientmsg_send(DltDaemonClientSockInfo* clientinfo,
+                              void* data1, int size1, void* data2, int size2, int verbose)
+{
+    PRINT_FUNCTION_VERBOSE(verbose);
+    if((clientinfo != NULL) && (clientinfo->isvalidflag == ADDRESS_VALID) &&
+        (data1 != NULL) && (size1 > 0) &&
+        (data2 != NULL) && (size2 > 0)) {
+        void *data = (void *) calloc(size1 + size2, sizeof(char));
+        if(data != NULL) {
+            memcpy(data, data1, size1);
+            memcpy(data + size1, data2, size2);
+
+            if(sendto(g_udp_sock_fd, data, size1 + size2, 0, (struct sockaddr*)&clientinfo->clientaddr,
+                    clientinfo->clientaddr_size) < 0)
+                dlt_log(LOG_ERR,"dlt_daemon_udp_clientmsg_send1: Send UDP Packet Data failed\n");
+
+            free(data);
+            data = NULL;
+        } else {
+            dlt_log(LOG_ERR,"dlt_daemon_udp_clientmsg_send: calloc failure\n");
```

Comment:
As proposed above, error case should go first and succeeded case afterwards.

Reply:
Done